Public/Src/Engine/Scheduler/Artifacts/FileContentManager.cs

```diff
@@ -334,6 +334,18 @@ public void ReportInputContent(FileArtifact artifact, in FileMaterializationInfo
             }
         }
 
+        /// <summary>
+        /// Report the change affected outputs the pip. Only report those produced by the pip.
+        /// </summary>
+        public void ReportSourceChangeAffectedOutputs(AbsolutePath output, PipOutputOrigin origin, bool isFile)
+        {
+            if (origin == PipOutputOrigin.Produced)
+            {
+                // Notify the host with content that was reported
+                m_host.ReportSourceChangeAffectedOutputs(output, origin, isFile);
+            }
+        }

```

Comment:
Why does this reporting have to go through `FileContentManager`?

Reply:
moved to IPipExecutionEnvironment.